test/integration/sequelize.test.js

```diff
@@ -386,6 +386,40 @@ describe(Support.getTestDialectTeaser('Sequelize'), function() {
       }).to.throw(Error, 'Both `replacements` and `bind` cannot be set at the same time');
     });
 
+    if (Support.dialectIsMySQL()) {
+      it.only('properly adds and escapes replacement value', function () {
+        var logSql,
+            number  = 1,
+            date = new Date(),
+            string = 't\'e"st',
+            boolean = true,
+            buffer = new Buffer('t\'e"st');
+
+        date.setMilliseconds(0);
+        return this.sequelize.query({
+            query: 'select ? as number, ? as date,? as string,? as boolean,? as buffer',
+            values: [number, date, string, boolean, buffer]
+          }, {
+            type: this.sequelize.QueryTypes.SELECT,
+            logging: function(s) {
+              logSql = s;
+            }
+          }).then(function(result) {
+            var res = result[0] || {};
+            res.date = res.date && new Date(res.date);
+            res.boolean = res.boolean && true;
+            expect(res).to.deep.equal({
+              number : number,
+              date   : date,
+              string : string,
+              boolean: boolean,
+              buffer : buffer
+            });
+            expect(logSql.indexOf('?')).to.equal(-1);
+        });
+      });
+    }
```

Comment:
There seem to be an issue with Buffers in Postgres where the return value
is the binsry representation of the string and not a Buffer.
```diff
 + expected - actual
       {
         "boolean": true
      -  "buffer": "\\x742765227374"
      +  "buffer": [Buffer: [
      +    116
      +    39
      +    101
      +    34
      +    115
      +    116
      +  ]]
         "date": [Date: 2016-02-10T08:29:33.000Z]
         "number": 1
         "string": "t'e\"st"
       }
```
I'm not sure if this is actual bug that need to be fixed and if this is caused because the replacement doesn't format buffer correctly or that the parsing of the result is the problem.
I can try and setup a local dev environment with postgres but it might take me some time